test/e2e/framework/resource/resource.go

```diff
@@ -0,0 +1,32 @@
+package resource
+
+import (
+	"github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/yaml"
+)
+
+type Resource struct {
+	gvr  schema.GroupVersionResource
+	ns   string
+	data []byte
+}
+
+func (r *Resource) Gvr() schema.GroupVersionResource { return r.gvr }
+func (r *Resource) Namespace() string                { return r.ns }
+func (r *Resource) Data() []byte                     { return r.data }
+func (r *Resource) IsClustered() bool                { return r.ns == "" }
+func (r *Resource) IsNamespaced() bool               { return !r.IsClustered() }
+
+func (r *Resource) Unstructured() *unstructured.Unstructured {
+	var u unstructured.Unstructured
+	gomega.Expect(yaml.Unmarshal(r.data, &u)).To(gomega.Succeed())
+	// TODO: set namespace ?
+	// TODO: ensure GV(R/K) ?
```

Comment:
What's pending here?

Reply:
I did not need it yet but it could be useful, at least for the namespace part.
Let's keep the TODOs for now, I guess if we need it it will pop up soon.